File: SWP.Core/Entities/Part.cs
using SWP.Core.GMSAttribute;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace SWP.Core.Entities
{
    [GMSTableName("part")]
    public class Part
    {
        [Column("part_id")]
        [Key]
        public int PartId { get; set; }

        [Column("part_name")]
        [Required]
        [MaxLength(100)]
        public string PartName { get; set; } = string.Empty;

        [Column("part_code")]
        [Required]
        [MaxLength(20)]
        public string PartCode { get; set; } = string.Empty;

        [Column("part_quantity")]
        [Required]
        public int PartQuantity { get; set; }

        [Column("part_unit")]
        [Required]
        [MaxLength(20)]
        public string PartUnit { get; set; } = string.Empty;

        [Column("part_category_id")]
        [Required]
        public int PartCategoryId { get; set; }

        [Column("part_price", TypeName = "decimal(24,2)")]
        public decimal? PartPrice { get; set; }

        [Column("warranty_month")]
        public int? WarrantyMonth { get; set; }

        [Column("is_deleted")]
        public int IsDeleted { get; set; } = 0;

        //
        public PartCategory PartCategory { get; set; } = null!;
        public ICollection<ServiceTicketDetail> ServiceTicketDetails { get; set; } = new List<ServiceTicketDetail>();
        public ICollection<PartStockOutItem> PartStockOutItems { get; set; } = new List<PartStockOutItem>();
        public ICollection<PartPriceHistory> PartPriceHistories { get; set; } = new List<PartPriceHistory>();
    }
}

File: GarageProject-BE-Springboot-MainFlow/src/main/java/fptu/edu/vn/training/entity/Part.java
package fptu.edu.vn.training.entity;

import fptu.edu.vn.training.enums.PartStatus;
import jakarta.persistence.CascadeType;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.OneToMany;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.util.LinkedHashSet;
import java.util.Set;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
@Entity
@Table(name = "part")
public class Part {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "part_id")
    private Integer partId;

    @Column(name = "part_name", nullable = false, length = 100)
    private String partName;

    @Column(name = "part_code", nullable = false, length = 20)
    private String partCode;

    @Column(name = "part_quantity", nullable = false)
    private Integer partQuantity;

    @Column(name = "part_unit", nullable = false, length = 20)
    private String partUnit;

    @Column(name = "part_category_id", nullable = false)
    private Integer partCategoryId;

    @Column(name = "part_price", precision = 24, scale = 2)
    private BigDecimal partPrice;

    @Column(name = "selling_price", precision = 24, scale = 2)
    private BigDecimal sellingPrice;

    @Enumerated(EnumType.STRING)
    @Column(name = "status", length = 20)
    private PartStatus status;

    @Column(name = "warranty_month")
    private Integer warrantyMonth;

    @Column(name = "is_deleted", nullable = false)
    @Builder.Default
    private Integer isDeleted = 0;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "part_category_id", insertable = false, updatable = false)
    private PartCategory partCategory;

    @Builder.Default
    @OneToMany(mappedBy = "part", fetch = FetchType.LAZY, cascade = CascadeType.ALL, orphanRemoval = false)
    private Set<ServiceTicketDetail> serviceTicketDetails = new LinkedHashSet<>();

    @Builder.Default
    @OneToMany(mappedBy = "part", fetch = FetchType.LAZY, cascade = CascadeType.ALL, orphanRemoval = false)
    private Set<PartStockOutItem> partStockOutItems = new LinkedHashSet<>();

    @Builder.Default
    @OneToMany(mappedBy = "part", fetch = FetchType.LAZY, cascade = CascadeType.ALL, orphanRemoval = false)
    private Set<PartPriceHistory> partPriceHistories = new LinkedHashSet<>();
}
